tests/test_2024_10_08.py

```diff
@@ -275,8 +277,47 @@ def validate_optimism_upgrade_call(
     ]
     validate_events_chain([e.name for e in events], _events_chain)
     assert events.count("SentMessage") == 1
+
+    assert events["SentMessage"]["target"] == target
+    assert events["SentMessage"]["sender"] == sender
+
     # Check bytecode that was sent to messenger to update L2 bridge and wstETH token
     encoded_l2_upgrade_call = encode_l2_upgrade_call(l2_bridge_proxy, l2_bridge_impl, l2_wst_proxy, l2_wst_impl)
     assert events["SentMessage"]["message"] == encoded_l2_upgrade_call
-    assert events["SentMessage"]["target"] == target
-    assert events["SentMessage"]["sender"] == sender
+
+    # Unparse and check l2 params
+    encoded_msg = encode_l2_message(events["SentMessage"]["message"])
```

Comment:
Maybe call it something like `decode_l2_upgrade_message`? To make it more explicit